torch/_inductor/lowering.py

```diff
@@ -5068,6 +5066,103 @@ def inner_fn_max_idx(idx):
     return rv, ri
 
 
+def pad_adaptive_loader_nd(x, spatial_dims, pad_val=0.0):
+    x_loader = x.make_loader()
+
+    def load(prefix, increments, start_indices, end_indices):
+        # Build mask for all spatial dimensions
+        mask = None
+        for i in range(spatial_dims):
+            dim_mask = ops.lt(
+                ops.index_expr(start_indices[i] + increments[i], torch.int64),
+                ops.index_expr(end_indices[i], torch.int64),
+            )
+            mask = dim_mask if mask is None else ops.and_(mask, dim_mask)
+
+        return ops.masked(
+            mask,
+            lambda: x_loader(
+                [
+                    *prefix,
+                    *(start_indices[i] + increments[i] for i in range(spatial_dims)),
+                ]
+            ),
+            pad_val,
+        )
+
+    return load
+
+
+def compute_indices_adaptive_pooling_nd(start_index, end_index, in_sizes, out_sizes):
+    start_fns = []
+    end_fns = []
+    for in_dim, out_dim in zip(in_sizes, out_sizes):
+        start_fns.append(
+            functools.partial(start_index, out_dim=out_dim, inp_dim=in_dim)
+        )
+        end_fns.append(functools.partial(end_index, out_dim=out_dim, inp_dim=in_dim))
+    return start_fns, end_fns
+
+
+def _adaptive_pooling_fn_nd(
```

Comment:
Why add `_adaptive_pooling_fn_nd` when `_adaptive_pooling_fn` can be generalized to n dimensions?

Reply:
I merged them.